### PR TITLE
fix event emitter leak warning

### PR DESCRIPTION
**Pull request by RB-Lab**
I have a really long integration test suite, that listens a lot on server edit messages. It was giving me these warnings:
```
 (node:103570) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11
 EditedMessageText listeners added to [TelegramServer]. Use emitter.setMaxListeners() to
 increase limit (Use `node --trace-warnings ...` to show where the warning was created)
```

### CODE REVIEW

File: src/telegramServer.ts [hunk 1]
```diff
@@ -217,29 +217,37 @@ export class TelegramServer extends EventEmitter {
...
-   * increase limit (Use `node --trace-warnings ...` to show where the warning was created)
-   */
   async waitBotEdits() {
     return new Promise<void>((resolve) => {
-      this.on('EditedMessageText', () => resolve());
+      const handler = () => {
+        this.off('EditedMessageText', handler);
+        resolve();
+      };
+      this.on('EditedMessageText', handler);
```
Comment (jehy) [inline src/telegramServer.ts:226]: I suppose we can simple use `once` onstead of `on` - the we don't have to remove listener manually.

File: src/telegramServer.ts [hunk 2]
```diff
@@ -217,29 +217,37 @@ export class TelegramServer extends EventEmitter {
...
+      this.on('AddedBotMessage', handler);
     });
   }
 
   async waitUserMessage() {
     return new Promise<void>((resolve) => {
-      this.on('AddedUserMessage', () => resolve());
-      this.on('AddedUserCommand', () => resolve());
-      this.on('AddedUserCallbackQuery', () => resolve());
+      const messageHandler = () => {
```
Comment (jehy) [inline src/telegramServer.ts:242]: That doesn't look cool but I can't think of a better way too :)

**Review by jehy — CHANGES_REQUESTED** [review]
I think that it would be a little better to use `once` in cases when we wait for one event.

**Comment by RB-Lab** [comment]
Good point!

File: src/telegramServer.ts [hunk 3]
```diff
@@ -217,29 +217,23 @@ export class TelegramServer extends EventEmitter {
...
+      this.once('AddedBotMessage', () => resolve());
     });
   }
 
   async waitUserMessage() {
     return new Promise<void>((resolve) => {
-      this.on('AddedUserMessage', () => resolve());
-      this.on('AddedUserCommand', () => resolve());
-      this.on('AddedUserCallbackQuery', () => resolve());
+      this.once('AddedUserMessage', () => resolve());
```
Comment (jehy) [inline src/telegramServer.ts:234]: Everything was right here in previous commit version. We wait for any signal and the turn off all emitters and resolve promise.

Now, one emitter will turn itself off but other two will stay and that's not what we expect :)

**Review by jehy — CHANGES_REQUESTED** [review]

File: src/telegramServer.ts (hunk 3 above)
Comment (RB-Lab) [reply src/telegramServer.ts:234]: Ahh, that's why you "can't think of a better way too"! =)) Indeed!

Comment (jehy) [reply src/telegramServer.ts]: Yup. Emitter is a great power which comes with great possibility of shooting your legs off :)

**Pull request merged**